### PR TITLE
feat(document-search): batch ingestion of documents

### PR DESCRIPTION
**Pull request by ludwiktrammer**

### CODE REVIEW

File: packages/ragbits-document-search/tests/unit/test_document_search.py [hunk 1]
```diff
@@ -163,3 +165,40 @@ async def test_document_search_ingest_multiple_from_sources():
...
+        DocumentMeta.create_text_document_from_literal("Name of Peppa's teacher is Madame Gazelle"),
+        DocumentMeta.create_text_document_from_literal("Name of Peppa's doctor is Dr. Brown Bear"),
+        DocumentMeta.create_text_document_from_literal("Name of Peppa's cousin is Chloe Pig"),
+        DocumentMeta.create_text_document_from_literal("Name of Peppa's cousin is Alexander Pig"),
+    ]
+
+    embeddings_mock = AsyncMock()
+    embeddings_mock.embed_text.return_value = [[0.1, 0.1]] * len(documents)
+
+    # processing_strategy = BatchedAsyncProcessing(batch_size=5)
```
Comment (mhordynski) [inline packages/ragbits-document-search/tests/unit/test_document_search.py:189]: I think it should be used here? Rather than sequential?

Comment (ludwiktrammer) [reply packages/ragbits-document-search/tests/unit/test_document_search.py:189]: That's exactly right. Thank you!

**Comment by mhordynski** [comment]
Let's add a how-to page how to set processing strategy / create a custom one

File: packages/ragbits-document-search/src/ragbits/document_search/ingestion/processor_strategies/batched.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+from ragbits.document_search.documents.document import Document, DocumentMeta
+from ragbits.document_search.documents.element import Element
+from ragbits.document_search.documents.sources import Source
+from ragbits.document_search.ingestion.document_processor import DocumentProcessorRouter
+from ragbits.document_search.ingestion.providers.base import BaseProvider
+
+from .base import ProcessingExecutionStrategy
+
+
+class BatchedAsyncProcessing(ProcessingExecutionStrategy):
```
Comment (kdziedzic68) [inline packages/ragbits-document-search/src/ragbits/document_search/ingestion/processor_strategies/batched.py:13]: question: have you measured the performance difference between batched and sequential strategies ?

File: packages/ragbits-document-search/src/ragbits/document_search/ingestion/processor_strategies/base.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+from typing_extensions import Self
+
+from ragbits.document_search.documents.document import Document, DocumentMeta
+from ragbits.document_search.documents.element import Element
+from ragbits.document_search.documents.sources import Source
+from ragbits.document_search.ingestion.document_processor import DocumentProcessorRouter
+from ragbits.document_search.ingestion.providers.base import BaseProvider
+
+
+class ProcessingExecutionStrategy(ABC):
```
Comment (kdziedzic68) [inline packages/ragbits-document-search/src/ragbits/document_search/ingestion/processor_strategies/base.py:13]: tbd: should we include the processing strategies in public api ?

**Review by mhordynski — APPROVED** [review]
lgtm

**Pull request merged**